src/module/terms-condition/terms-condition.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from "@nestjs/common";
+import { InjectModel } from "@nestjs/mongoose";
+import { Model } from "mongoose";
+import { TermsCondition } from "./terms-condition.schema";
+
+
+@Injectable()
+export class TermsConditionService {
+    constructor(
+        @InjectModel('TermsCondition') private readonly termsConditionModel: Model<TermsCondition>
+    ) { }
+
+    async createTermsCondition(content: string): Promise<TermsCondition> {
+
+        const latest = await this.termsConditionModel.findOne({ active: true }).sort({ version: -1 })
+
+        if (latest) {
+            latest.active = false;
+            await latest.save();
+        }
+
+        const newEntry = new this.termsConditionModel({ content, version: latest ? latest.version + 1 : 1, active: true });
```

Comment:
for generating version add more details about the version

like add year + month + date 

` const date = new Date();
    const version = `v${date.getFullYear()}.${date.getMonth() + 1}.${date.getDate()}`;`